dali/base/dadfs.cpp

```diff
@@ -7929,6 +7930,130 @@ IDistributedSuperFile *CDistributedFileDirectory::createSuperFile(const char *_l
 }
 
 // MORE: This should be implemented in DFSAccess later on
+IDistributedSuperFile *CDistributedFileDirectory::createSuperFile(IPropertyTree *tree, IUserDescriptor *user, bool interleaved, bool ifdoesnotexist, IDistributedFileTransaction *transaction)
+{
+    StringBuffer superName;
+    Owned<IDistributedSuperFile> superfile;
+
+    if (!tree->hasProp("@name"))
+        throw MakeStringException(-1, "createSuperFile: There is no SuperFile name in XML");
+
+    tree->getProp("@name", superName);
+    if (superName.isEmpty())
+        throw MakeStringException(-1, "createSuperFile: SuperFile name is empty in XML");
+
+    const char *pSuperName = superName.str();
+    if (queryDistributedFileDirectory().exists(pSuperName, user, false, true))
+        throw MakeStringException(-1, "createSuperFile: SuperFile '%s' already exists!", pSuperName);
+
+    // Check Physical Parts of SubFiles
+    unsigned numSubfiles = tree->getPropInt("@numsubfiles", 0);
+    if (numSubfiles)
+    {
+        Owned<IPropertyTreeIterator> treeIter = tree->getElements("SubFile");
+        ForEach(*treeIter)
+        {
+            IPropertyTree & st = treeIter->get();
+            StringBuffer subfilename;
+            st.getProp("@name", subfilename);
+            if (!queryDistributedFileDirectory().existsPhysical(subfilename.str(), user))
+                throw MakeStringException(-1, "createSuperFile: Physical Parts of '%s' not found!", subfilename.str());
+        }
+    }
+    // It seems XML is fine add this superfile to root tree
+
+    // Get the correct branch name
+    StringBuffer branchName;
+    branchName.set(queryDfsXmlBranchName(DXB_SuperFile)).trim();
+    const char *branch = branchName.str();
+
+    // Get the head path and connect with it
+    CDfsLogicalFileName sdlname;
+    sdlname.set(pSuperName);
+    StringBuffer head;
+    sdlname.makeScopeQuery(head.clear(), true);
+    Owned<IRemoteConnection> conn = querySDS().connect(head.str(), myProcessSession(), 0, FOREIGN_DALI_TIMEOUT);
```

Comment:
More significantly, this isn't the approach we discussed.

Creating the IDistributedSuperFile should be akin to creating a IDistributeFile (see IDistributedFileDirectory::createNew). 
It should create and a IDistributedSuperFile, without committing changes to Dali.
Then you can do a super->attach() to attach it to Dali.
Verification of subfiles/consistentcy etc. should as much as possible use existing CDistributedSuperFile methods.